concrete/src/Page/Stack/Stack.php

```diff
@@ -54,24 +55,47 @@ public static function mapImportTextToType($type)
      *
      * @param Collection $collection
      * @param string $arHandle
-     * @return void
+     * @return Stack|null
+     * @throws \Exception
      */
     public static function getGlobalAreaStackFromName(Collection $collection, string $arHandle): ?Stack
     {
         $app = Application::getFacadeApplication();
         $db = $app->make(Connection::class);
         $checker = new Checker($collection);
-        
+
         /** @var \Concrete\Core\Cache\Level\RequestCache $requestCache */
         $requestCache = $app->make('cache/request');
-        $identifier = sprintf('/stack/global_area/%s/cID', $arHandle);
+
+        /** @var MultilingualDetector $md */
+        $md = $app->make(MultilingualDetector::class);
+        if ($md->isEnabled()) {
+            $ps = $md->getPreferredSection();
+            $psID = $ps ? $ps->getCollectionID() : 0;
+            $identifier = sprintf('/stack/global_area/%s/%s/cID', $arHandle, $psID);
+        } else {
+            $identifier = sprintf('/stack/global_area/%s/cID', $arHandle);
+        }
         $item = $requestCache->getItem($identifier);
         if ($item->isHit()) {
             $stackID = $item->get();
         } else {
-            $stackID = $db->executeQuery('select cID from Stacks where stName = ? and stType = ?', [
-                $arHandle, self::ST_TYPE_GLOBAL_AREA
-            ])->fetchOne();
+            $qb = $db->createQueryBuilder();
+            $q = $qb->select('cID')
+                ->from('Stacks', 's')
+                ->andWhere('s.stName = :stName')
+                ->andWhere('s.stType = :stType')
+                ->setParameter('stName', $arHandle)
+                ->setParameter('stType', self::ST_TYPE_GLOBAL_AREA);
+
+            if ($md->isEnabled()) {
+                $ps = $md->getPreferredSection();
+                $psID = $ps ? $ps->getCollectionID() : 0;
+                $q->andWhere('s.stMultilingualSection = :stMultilingualSection');
```

Comment:
This is not correct: if a stack/global area doesn't have a localized version, we should use the "language-neutral" one